### ISSUE TITLE
Field-level CSS class attribute

### ISSUE THREAD

**Issue by SachaG** [aldeed/meteor-autoform#869]
It'd be useful to have the ability to style specific fields differently. So I was thinking maybe we could have a `autoform.cssClass` attribute in the schema that simply applies a class to a field's container element?

**Comment by zimme** [comment]
```js
Schemas.login = new SimpleSchema({
  email: {
    autoForm: {
      class: 'custom classes'
    },
    regEx: SimpleSchema.regEx.Email,
    type: String
  }.
  password: {...}
});
```

Doesn't this work?

**Comment by aldeed** [comment]
Yeah, any attribute can be in `autoform` and just as if it were in the markup, it will be passed along to the rendered element.

So:

```js
{
  autoform: {
    afFieldInput: {
      class: 'custom'
    }
  }
}
```

It will also work to do `{autoform: {class: 'custom'}}` without the `afFieldInput`, but if you don't specify which component the attributes are for, then they're added to all components, which may not be what you want.

If you're referring to adding a class to the form group div, then I agree we should add some way to do that. I think there's already an issue for it, but I could be wrong. My thought was to look for `div-` prefixed attributes on `afFormGroup` and add all of those to the div. This would need to be implemented by each afFormGroup theme template, but we may also need some logic in the core to ignore the attributes.

**Comment by SachaG** [comment]
Oh I wasn't aware of that feature, I'll try it.

**Comment by zimme** [comment]
Can't we just do this?
```js
{
  autoform: {
    afFormGroup: {
      class: 'custom'
    }
  }
}
```

**Comment by aldeed** [comment]
@zimme, no because attributes on the form group are forwarded to the `afFieldInput` within it, unless they are prefixed with `label-`, in which case they're added to the label element. So we could do the same thing with `div-`. I think it makes sense this way because most commonly you are expecting the extra attributes to end up on the input element. That might be just me, and I'm open to debating it.

**Comment by zimme** [comment]
Aah, I see. Just tried it out.

My first thought was maybe use `input-` but then I thought `div-` might be better and make `class` work as it do today.

Another approach might be this:
```js
{
  autoform: {
    afFormGroup: {
      label: {
        class: 'custom-label-class'
      },
      input: {
        class: 'custom-input-class',
      },
      class: 'custom-div-class'
    }
  }
}
```

**Comment by SachaG** [comment]
I'm not sure if the ability to assign a class to an inner element is really needed. If you could assign a class to the outer container, then you'd just be able to target anything with `.myClass input` or `.myClass label`?

**Comment by zimme** [comment]
Assign classes to labels is useful.

Say you use bootstrap you can assign 'sr-only' to a label to make sure it only shows when necessary.

@aldeed, I tried to use `label-class` in `{{> quickForm}}` but I saw it was only available for `bootstrap-horizontal` theme.

**Comment by aldeed** [comment]
@zimme, yeah `label-class` is only for setting the column classes on horizontal theme. But `label-anything` works on `afQuickField` or `afFormGroup`. You can also use a component name prefix on quickForm to quickly set for everything within that form, so `afFormGroup-label-class="foo"` on a `quickForm` might work, though I'm not sure I ever tested it.

@SachaG, there might be attributes other than class that people want on inner elements, like `data-` attributes, etc.

**Comment by zimme** [comment]
Nice one, didn't know about that... I'm gonna try that someday, for now I just used the schema autoform definitions which is nice

**Comment by elGusto** [comment]
This issue and this pull request are about the same thing: https://github.com/aldeed/meteor-autoform/pull/939

**Comment by TimLCooley** [comment]
Would it be possible to get an example of how this works with a bootstrap class.  for example a 
class="col-md-6."  I have been trying to get this to work and I keep breaking it some place.  This is the code.  from what II understand I need to do Step 2
Step 1
name: {
        type: String,
        label: "Recipe Name",
        max: 200
    },


Step 2
    name: {
              afFieldInput: {
              class: 'col-md-6'
         },
        type: String,
        label: "Recipe Name",
        max: 200
    },

**Comment by aldeed** [comment]
I've looked at the PR and will merge it once we iron out a few issues.

@TimLCooley The bootstrap horizontal templates look for special attribute `input-col-class` on the autoForm/quickForm.

**Comment by fknipp** [comment]
@aldeed has already merged #939. This is the code I used in my schema definition to show fields side-by-side. It works with the templates `bootstrap3` as well as with `bootstrap3-horizontal`.

```coffee
Schemas.Registrations = new SimpleSchema
        'participant':
                type: Object
                autoform:
                        afObjectField:
                                bodyClass: 'container-fluid row'
        'participant.firstname':
                type: String
                max: 60
                autoform:
                        afFormGroup:
                                'formgroup-class': 'col-sm-6'
        'participant.lastname':
                type: String
                max: 60
                autoform:
                        afFormGroup:
                                'formgroup-class': 'col-sm-6'
```

**Comment by bySabi** [comment]
Hi @zimme how you add a class to label tag on a formGroup on schema definition?

I just want show fontawesome glyphs in place of label, say:

```
var home = new SimpleSchema({
  home: {
    type: String,
    autoform: {

      // this not work!!!
      // I need add class 'fa' to  <label class="col-sm-3 text-right control-label" ... 
      'afFormGroup-label-class': 'fa',

      label: '\uf015'       //Unicode 'home' glyph
    }
  }
});
```

Thanks!
